python/composio/tools/toolset.py

```diff
@@ -1405,14 +1404,9 @@ def initiate_connection(
                     ),
                     auth_scheme=auth_scheme,
                 ).id
-            except NoItemsFound as e:
-                raise ComposioSDKError(
-                    message=(
-                        f"No existing integration found for `{str(app)}`, "
-                        "Please create an integration and use the ID to "
-                        "initiate connection."
-                    )
-                ) from e
+            except NoItemsFound:
+                integration = self.create_integration(app=app, auth_mode=auth_scheme)
```

Comment:
Consider adding error handling for the `create_integration` call to provide better error messages:
```python
try:
    integration = self.create_integration(app=app, auth_mode=auth_scheme)
    integration_id = integration.id
except Exception as e:
    raise ComposioSDKError(
        f"Failed to auto-create integration for {app}: {str(e)}"
    ) from e
```